test/lib/modules/dataStore.test.js

```diff
@@ -926,6 +1150,30 @@ describe('dataStore', () => {
         return done();
       });
     });
+
+    it('should resolve with a query object', (done) => {
+      nock('https://baas.kinvey.com')
+        .matchHeader('content-type', 'application/json')
+        .matchHeader('x-kinvey-api-version', '3')
+        .matchHeader('x-kinvey-skip-business-logic', 'true')
+        .delete(`/appdata/${environmentId}/myCollection/`)
+        .basicAuth({
+          user: environmentId,
+          pass: mastersecret
+        })
+        .query({ query: '{"foo":"bar"}' })
+        .reply(200, { count: 12 });
+
+      const query = new Query();
+      query.equalTo('foo', 'bar');
+
+      const collection = this.store().collection('myCollection');
+      collection.remove(query)
+        .then((result) => {
+          result.should.containDeep({ count: 12 });
```

Comment:
Nit: I think it's a good idea to auto-generate sample data (vs using hard-coded values) as an extra layer of protection